buildlib/azure-pipelines.yml

```diff
@@ -0,0 +1,94 @@
+# See https://aka.ms/yaml
+
+trigger:
+  - master
+  - v1.6.x
+pr:
+  - master
+  - v1.6.x
+
+resources:
+  containers:
+    - container: centos7
+      image: ucfconsort.azurecr.io/ucx/centos7:1
+      endpoint: ucfconsort_registry
+    - container: fedora
+      image: ucfconsort.azurecr.io/ucx/fedora:30
+      endpoint: ucfconsort_registry
+
+stages:
+  - stage: Build
+    jobs:
+      - job: latest_cc
+        displayName: Latest CCs and CppCheck
+        container: fedora
+        steps:
+          - bash: ./autogen.sh
+            displayName: Setup autotools
+
+          - bash: |
+              mkdir build-gcc && cd build-gcc
+              # cscppc wraps gcc to use its output for cppcheck
+              export PATH="`cscppc --print-path-to-wrap`:$PATH"
+              ../contrib/configure-release
+              make -j`nproc` V=1 2>&1 | tee cc.log
+            displayName: GCC
+
+          - bash: |
+              cd build-gcc
+              cppcheck --version
+
+              cppcheck_err="cppcheck.err"
+              # use cs* tools to keep only UCX related issues
+              cslinker --quiet cc.log \
+                | csgrep --mode=json --path $(dirname $PWD) --strip-path-prefix $(dirname $PWD) \
+                | csgrep --mode=json --invert-match --path 'conftest.c' \
+                | csgrep --mode=grep --invert-match --event "internal warning" --prune-events=1 \
+                > $cppcheck_err
+
+                if [ -s $cppcheck_err ]; then
+                  echo "CppCheck found errors:"
+                  cat $f
```

Comment:
where is 'f' variable set?
maybe print cppcheck log anyway (in case of success as well) to show that it ran?

Reply:
The log will be empty, nothing to print.
I've updated the original PR, will update this one once that is merged.

Reply:
maybe remove --quiet parameter to cslinker?

Reply:
From its help: `q [ --quiet ]        do not report non-fatal errors`.
Not sure what is meant by "non-fatal errors" since it reports `cppcheck`'s warnings along with errors, but it is what RH guys use.

Reply:
ok